api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -144,4 +144,22 @@ public int compareTo(CohortMembership o) {
 		}
 		return ret;
 	}
+	
+	/**
+	 * @since 2.3.0
+	 * Indicates if a given cohortMembership object is equal to this one
+	 * 
+	 * @param otherCohortMembership is a CohortMembership object that should be checked for equality with this object
+	 * @return true if both objects are logically equal. This is the case when endDate, startDate and patientId are equal  
+	 */
+	public boolean equals(CohortMembership otherCohortMembership) {
+		if(this == otherCohortMembership) return true;
+		
+		return otherCohortMembership != null && 
+			((endDate != null ) ? this.endDate.equals(otherCohortMembership.getEndDate()) : otherCohortMembership.getEndDate() == null)
```

Comment:
Any reason why you sometimes use endDate and others you use this.endDate?
At first appearance, it looked as if endDate is a local variable different from this.endDate

Reply:
"this." is not necessary since only the class attributes exist with this name (no local variable). 
At first I thought it would make it more obvious that "this" class member is checked for equality with the attribute. But since the statement is already quite long to avoid NullPointerExceptions I think the better way to go is to remove the "this" in this method.

This was done in the new commit